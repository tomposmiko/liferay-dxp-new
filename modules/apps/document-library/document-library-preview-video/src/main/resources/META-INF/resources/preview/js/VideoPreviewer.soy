{namespace com.liferay.document.library.preview.VideoPreviewer}

/**
 * VideoPreviewer Component
 */
{template .render}
	{@param videoSources: list<?>}
	{@param? id: string}
	{@param? videoPosterURL: string}

	<div class="preview-file" id="{$id}">
		<div class="preview-file-container preview-file-max-height">
			<video
				class="preview-file-video"
				controls
				controlsList="nodownload"
				{if $videoPosterURL}
					poster="{$videoPosterURL}"
				{/if}
			>
				{foreach $videoSource in $videoSources}
					<source src="{$videoSource.url}" type="{$videoSource.type}">
				{/foreach}
			</video>
		</div>
	</div>
{/template}