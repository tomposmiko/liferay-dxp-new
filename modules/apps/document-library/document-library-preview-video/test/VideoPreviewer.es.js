/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import VideoPreviewer from '../src/main/resources/META-INF/resources/preview/js/VideoPreviewer.es';

let component;

describe('document-library-preview-video', () => {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('renders a video player', () => {
		component = new VideoPreviewer({
			videoPosterURL: 'poster.jpg',
			videoSources: [
				{
					type: 'video/mp4',
					url: '//video.mp4'
				},
				{
					type: 'video/ogv',
					url: '//video.ogv'
				}
			]
		});

		expect(component).toMatchSnapshot();
	});
});
