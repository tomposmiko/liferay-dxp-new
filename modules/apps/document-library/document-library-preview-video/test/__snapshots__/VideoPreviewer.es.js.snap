// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document-library-preview-video renders a video player 1`] = `
<div class="preview-file">
  <div class="preview-file-container preview-file-max-height"><video class="preview-file-video" controls="" controlslist="nodownload" poster="poster.jpg">
      <source src="//video.mp4" type="video/mp4">
      <source src="//video.ogv" type="video/ogv"></video></div>
</div>
`;
